### PR TITLE
Add Initial Mailbox panel and sensor

### PR DESCRIPTION
**Pull request by PhracturedBlue**
This is a new panel that adds an interface to and Asterisk voice-mailbox

The  panel allows viewing messages, reading a speech-to-text translation of the messages, listening to the messages, and deleting messages

I've cleaned up the code now, and gotten the asynchronous MP3 loading working now

### CODE REVIEW

File: panels/mailbox/ha-panel-mailbox.html [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+<link rel="import" href="../../bower_components/vaadin-combo-box/vaadin-combo-box.html">
+
+<link rel='import' href='../../src/components/ha-menu-button.html'>
+<link rel='import' href='../../src/resources/ha-style.html'>
+
+<link rel="import" href="../../bower_components/vaadin-grid/vaadin-grid.html">
+<link rel="import" href="../../bower_components/vaadin-grid/vaadin-grid-selection-column.html">
+
+
+<dom-module id='ha-panel-mailbox'>
```
Comment (balloob) [inline panels/mailbox/ha-panel-mailbox.html]: Since it's bound to `asteriskmbox`, can you name it `ha-panel-asterisk` ?

**Comment by balloob** [comment]
- Please rebase to fix the merge conflict
- Please address lint errors in the Travis logs

File: panels/mailbox/ha-panel-mailbox.html [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    var grid = this.$.msgtable;
+    var items = grid.selectedItems;
+    var len = items.length;
+    for (var i = 0; i < len; i++) {
+      var platform = items[i].platform;
+      this.hass.callApi('DELETE', 'mailbox/delete/' + platform + '/' + items[i].sha);
+    }
+  },
+  getMessages: function () {
+    let items = this.platforms.map((platform) => {
```
Comment (balloob) [inline panels/mailbox/ha-panel-mailbox.html]: You're lucky, we just enabled es2015 compilation so this is ok now.

**Comment by balloob** [comment]
Screenshots:

image

image

**Comment by balloob** [comment]
You can't have the message details as a card inside the grid, as that is also represented as a card. Can you move it to the listen dialog. The listen dialog can contain the text, the audio and the delete button. That way you can also get rid of the select row functionality.

I would actually move the list to be rendered using [`paper-item`](https://www.webcomponents.org/element/PolymerElements/paper-item) inside a paper-card, with the dialog popping open on tap.

Quick mockup:
image

**Comment by balloob** [comment]
Can you also add the `with-backdrop` attribute to the dialog and make it responsive like we do with our other dialogs https://github.com/home-assistant/home-assistant-polymer/blob/master/src/dialogs/ha-voice-command-dialog.html#L82-L104

**Comment by PhracturedBlue** [comment]
I'm fine with moving the delete to the popup, but is there a way I can keep the message text on the main page (at least when on a larger display)?  If there are multiple messages, it is very convenient to be able to scan through them with a short description of each one, without having to pop-open each message and see.

I also don't understand what you mean by using a card within a card.  As far as I can tell, I followed the example of using the vaadin-grid object to add the message details.

**Comment by PhracturedBlue** [comment]
I think I have a reasonable solution now using paper-item.  However, I only want to show the 1st two lines of the message in the paper-item (which would align with the material-design concept of no-more than 3 lines per item).  I have tried a lot of different methods to achieve that, but I have been unsuccessful so far.  Any ideas?

**Comment by PhracturedBlue** [comment]
FYI,
I added the following pull request to make it easier to see he string translation:

https://github.com/home-assistant/home-assistant/pull/8893

**Comment by balloob** [comment]
image

Please add a `max-width: 600px` and a `margin: 0 auto` to `.content` to make it centered.

You can't put a header in a paper-item. That doesn't work. Have a look at the spec: https://material.io/guidelines/components/lists.html#lists-specs

- Text has to be left aligned
- No bold text on the second line.
- Don't justify the snippet

The frontend strictly follows the material design guidelines. There is no room for artistic freedom as consistency is more important than one list slightly being fitted for a better use case.

**Comment by PhracturedBlue** [comment]
Bold text on the 2nd line seems to be allowed:
image
I don't see an example without the avatar, but I also don't see anything in the specs that this mechanism is only for 3-lines with avatar

**Comment by PhracturedBlue** [comment]
It doesn't look quite as nice in my opinion, but I think it matches the material-design spec now.

**Comment by balloob** [comment]
I polished it up.

image

image

**Comment by balloob** [comment]
More polishing could be done. Time of message could be swapped with date, probably makes more sense. Also use a smaller date format.

**Comment by PhracturedBlue** [comment]
Thank you for all the patience and help you've supplied helping me to get this feature merged.  It is really appreciated.

**Comment by PhracturedBlue** [comment]
Is there a preferred date format?
Something like: 2017/08/08-12:22PM

Or: Aug 8 2017 10:05AM
Are dates showed elsewhere on HASS that I should be consistent to?

**Comment by balloob** [comment]
We format them based on the user preference in their browser. Just looked, we actually don't have a short format right now

**Pull request merged**